evm/15_core-concepts/46_EOAs-vs-Smart-Contracts.md

```diff
@@ -0,0 +1,85 @@
+
+### What are Smart Contracts?
+Smart contracts are self-executing contracts with predefined rules and conditions. They are autonomous pieces of code that reside on the blockchain and execute when triggered by specific transactions or events. Smart contracts on EOSEVM are typically written in Solidity or Vyper programming languages.
+
+#### What can you do with a Smart Contract?
+Smart contracts enable various functionalities and applications, including:
+
+- Creating decentralized applications (dApps) that run on the blockchain.
+- Implementing complex business logic and automating processes.
+- Defining and enforcing rules for digital assets, such as tokens or non-fungible assets.
+- Enabling secure and transparent transactions without intermediaries.
+- Interacting with other smart contracts and external systems.
+
+#### What you cant do with a Smart Contract?
+Although powerful, there are limitations to what smart contracts can do:
+
+- Smart contracts cannot access external data sources directly; they rely on oracles to fetch external information.
+- They cannot perform actions that require off-chain interactions, such as interacting with the internet or making API calls.
+- Smart contracts cannot initiate transactions on their own; they need to be triggered by external accounts (EOAs).
+
+#### How do you make a smart contract?
+To create a smart contract, you typically follow these steps:
+
+Choose a programming language: Solidity and Vyper are popular choices for writing smart contracts on EOSEVM.
+Write the contract code: Use a text editor or an integrated development environment (IDE) like Remix to write the code of your smart contract.
+Here's an example of a simple smart contract written in Solidity:
+
+```
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract SimpleContract {
+    string public message;
+
+    constructor(string memory initialMessage) {
+        message = initialMessage;
+    }
+
+    function setMessage(string memory newMessage) public {
+        message = newMessage;
+    }
+}
+```
+
+Compile the code: Use a compiler specific to the programming language to convert your smart contract code into bytecode, which is understood by the EOSEVM.
+Deploy the contract: Use a wallet or a deployment tool to deploy the smart contract bytecode onto the EOSEVM network.
+Interact with the smart contract: Once deployed, EOAs can interact with the smart contract by calling its functions and sending transactions.
+
+### How are EOAs and Smart contracts different?
+EOAs and smart contracts are distinct entities on the EOSEVM network:
+
+- EOAs represent individual user accounts and hold balances, while smart contracts are pieces of code that reside on the blockchain and execute predefined logic.
```

Comment:
The way this reads makes it sound like smart contracts can't hold balances too

Reply:
ok.